### PR TITLE
Support :const

### LINKED ISSUE #114: const is missing
```
const a = 1
```

### PR DESCRIPTION
**Pull request by tyru**
Fix #114

### CODE REVIEW

File: autoload/vimlparser.vim [hunk 1]
```diff
@@ -1865,6 +1905,22 @@ function! s:VimLParser.parse_lvalue()
...
+" TODO: merge with s:VimLParser.parse_lvalue()
+function! s:VimLParser.parse_constlvalue()
+  let p = s:LvalueParser.new(self.reader)
+  let node = p.parse()
+  if node.type == s:NODE_IDENTIFIER
+    if !s:isvarname(node.value)
+      throw s:Err(printf('E461: Illegal variable name: %s', node.value), node.pos)
+    endif
+  endif
+  if node.type == s:NODE_IDENTIFIER || node.type == s:NODE_CURLYNAME
```
Comment (tyru) [inline autoload/vimlparser.vim]: In contrast to `s:VimLParser.parse_lvalue()`, this does not handle

* `s:NODE_SUBSCRIPT` (`dict["key"]`)
* `s:NODE_SLICE` (`list[a:b]`)
* `s:NODE_DOT` (`dict.key`)
* `s:NODE_OPTION` (`&expandtab`)
* `s:NODE_ENV` (`$VAR`)
* `s:NODE_REG` (`@a`)

```
  if node.type == s:NODE_IDENTIFIER || node.type == s:NODE_CURLYNAME || node.type == s:NODE_SUBSCRIPT || node.type == s:NODE_SLICE || node.type == s:NODE_DOT || node.type == s:NODE_OPTION || node.type == s:NODE_ENV || node.type == s:NODE_REG
```

so `throw s:Err('Invalid Expression', node.pos)` will be executed.

**Comment by tyru** [comment]
@mattn please review this

File: autoload/vimlparser.vim (hunk 1 above)
Comment (mattn) [reply autoload/vimlparser.vim]: Do you have plan to sparate error message for E996?

```
let a = [0]
const a[0] = 1 " E996: Cannot lock a list or dict
let a = {}
const a["b"] = 1 " E996: Cannot lock a list or dict
const a.b = 1 " E996: Cannot lock a list or dict
const $a = 1 " E996: Cannot lock an environment variable
const @a = 1 " E996: Cannot lock a register
const &encoding = "foo" " E996: Cannot lock an option
```

Comment (tyru) [reply autoload/vimlparser.vim]: Sure. certainly it is better to throw E996 error here (from vimlparser).
I'll fix it.

**Comment by tyru** [comment]
> The job exceeded the maximum log length, and has been terminated.

only in "Vim master" 🤔 
https://travis-ci.org/vim-jp/vim-vimlparser/jobs/563346598
But "Vim master" output is 1371 lines, "Vim v8" output is 1435 lines? 🤔 

@blueyed do you know how to resolve this?

**Comment by blueyed** [comment]
> > The job exceeded the maximum log length, and has been terminated.
> 
> only in "Vim master" 
> https://travis-ci.org/vim-jp/vim-vimlparser/jobs/563346598
> But "Vim master" output is 1371 lines, "Vim v8" output is 1435 lines? 

Very odd.  I've restarted them, but since the PR and push build failed, it likely will not help.

Maybe some special character that triggers this?
Normally it will allow like 10k lines of output, and no activity for 10 minutes.

**Comment by tyru** [comment]
restarted. it's all green now.
@mattn please review 🙏

**Comment by tyru** [comment]
Thanks! 🎉

**Pull request merged**